test/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests/RunnableProjectGeneratorTests.cs

```diff
@@ -0,0 +1,124 @@
+using Microsoft.TemplateEngine.TestHelper;
+using Xunit;
+using static Microsoft.TemplateEngine.Orchestrator.RunnableProjects.RunnableProjectGenerator;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects.UnitTests
+{
+    public class RunnableProjectGeneratorTests : IClassFixture<EnvironmentSettingsHelper>
+    {
+        private EnvironmentSettingsHelper _environmentSettingsHelper;
+
+        public RunnableProjectGeneratorTests(EnvironmentSettingsHelper environmentSettingsHelper)
+        {
+            _environmentSettingsHelper = environmentSettingsHelper;
+        }
+
+        [Fact]
+        public async void CreateAsyncTest_GuidsMacroProcessingCaseSensitivity()
+        {
+            //
+            // Template content preparation
+            //
+
+            Guid inputTestGuid = new Guid("12aa8f4e-a4aa-4ac1-927c-94cb99485ef1");
+            string contentFileNamePrefix = "content - ";
+
+            string guidsConfigFormat = @"
+{{
+  ""guids"": [
+    ""{0}"" 
+  ]
+}}
+";
+            string guidsConfig = string.Format(guidsConfigFormat, inputTestGuid);
+
+            IDictionary<string, string?> templateSourceFiles = new Dictionary<string, string?>();
+            // template.json
+            templateSourceFiles.Add(TemplateConfigTestHelpers.DefaultConfigRelativePath, guidsConfig);
+
+            //content
+            foreach (string guidFormat in GuidMacroConfig.DefaultFormats.Select(c => c.ToString()))
+            {
+                templateSourceFiles.Add(contentFileNamePrefix + guidFormat, inputTestGuid.ToString(guidFormat));
+            }
+
+            //
+            // Dependencies preparation and mounting
+            //
+
+            IEngineEnvironmentSettings environment = _environmentSettingsHelper.CreateEnvironment();
+            string sourceBasePath = FileSystemHelpers.GetNewVirtualizedPath(environment);
+
+            foreach (KeyValuePair<string, string?> fileInfo in templateSourceFiles)
+            {
+                string filePath = Path.Combine(sourceBasePath, fileInfo.Key);
+                string fullPathDir = Path.GetDirectoryName(filePath)!;
+                environment.Host.FileSystem.CreateDirectory(fullPathDir);
+                environment.Host.FileSystem.WriteAllText(filePath, fileInfo.Value ?? string.Empty);
+            }
+
+            string targetDir = FileSystemHelpers.GetNewVirtualizedPath(environment);
+
+            //create mount point
+            IMountPoint? sourceMountPoint = null;
+            foreach (var factory in environment.Components.OfType<IMountPointFactory>())
+            {
+                if (factory.TryMount(environment, null, sourceBasePath, out IMountPoint myMountPoint))
+                {
+                    sourceMountPoint = myMountPoint;
+                    break;
+                }
+            }
+            Assert.True(sourceMountPoint != null, "couldn't create source mount point");
```

Comment:
nit: I think there was a helper method doing it. imo, mount point is ok to be extracted to method.